framework/src/userobjects/GeometryBase.C

```diff
@@ -58,4 +63,19 @@ GeometryBase::meshChanged()
       snapNode(node);
     }
   }
+
+  // go over blocks
+  MeshBase::node_iterator node = mesh.active_nodes_begin();
+  MeshBase::node_iterator node_end = mesh.active_nodes_end();
+  for (; node != node_end; ++node)
+  {
+    // check if node is part of any of the selected blocks
+    const auto & node_blocks = _mesh.getNodeBlockIds(**node);
+    for (const auto subdomain_id : _subdomain_ids)
+      if (node_blocks.count(subdomain_id))
+      {
+        snapNode(**node);
+        break;
+      }
+  }
```

Comment:
Should we be checking whether blocks are lower dimensional? Maybe I just need to know what we're snapping to. I guess in my brain I'm visualizing snapping all the nodes of a QUAD or TRI onto a line which doesn't sound good

Reply:
Yeah, I was thinking about that, too, but the API does permit snapping to a volume, not just a surface. This you enforce  proper mesh biasing upon refinement for example.